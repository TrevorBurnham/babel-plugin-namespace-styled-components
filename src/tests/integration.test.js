import path from 'path';
import React from 'react';
import renderer from 'react-test-renderer';
import { transformFileSync } from 'babel-core';
import 'jest-styled-components';

const evalFixture = filename => {
  const { code } = transformFileSync(filename, {
    plugins: [
      [path.join(__dirname, '../index.js'), { namespace: '.namespace' }],
    ],
  });

  if (code == null) throw new Error(`Fixture not found: ${filename}`);

  return eval(code);
};

describe('styled-components output', () => {
  test('for a style block with no selectors', () => {
    const Simple = evalFixture(path.join(__dirname, 'fixtures/simple.js'));
    expect(
      renderer.create(<Simple backgroundColor="#333" />).toJSON()
    ).toMatchSnapshot();
  });

  test('for a style block with &&', () => {
    const Input = evalFixture(
      path.join(__dirname, 'fixtures/doubleAmpersand.js')
    );
    expect(
      renderer
        .create(<Input borderWidth="1px" borderColor="fuchsia" />)
        .toJSON()
    ).toMatchSnapshot();
  });

  test('for a style block with a sibling selector', () => {
    const Button = evalFixture(
      path.join(__dirname, 'fixtures/siblingSelector.js')
    );
    expect(
      renderer.create(<Button padding="4px" spaceBetween="8px" />).toJSON()
    ).toMatchSnapshot();
  });

  test('for a style block with interpolated selectors', () => {
    const Parent = evalFixture(
      path.join(__dirname, 'fixtures/interpolatedSelector.js')
    );
    expect(
      renderer
        .create(
          <Parent childStyles="transform: scale(90%);" spaceBetween="12px" />
        )
        .toJSON()
    ).toMatchSnapshot();
  });

  test('for a style block with interpolated mixins', () => {
    const Menu = evalFixture(
      path.join(__dirname, 'fixtures/interpolatedMixin.js')
    );
    expect(renderer.create(<Menu />).toJSON()).toMatchSnapshot();
  });

  test('for a style block with interpolations before and afer a selector', () => {
    const Span = evalFixture(
      path.join(__dirname, 'fixtures/interpolationSandwich.js')
    );
    expect(renderer.create(<Span />).toJSON()).toMatchSnapshot();
  });
});
