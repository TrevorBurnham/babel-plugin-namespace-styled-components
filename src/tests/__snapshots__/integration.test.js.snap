// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled-components output for a style block with && 1`] = `
.namespace .c0.c0 {
  border: 1px solid fuchsia;
}

<input
  className="c0"
/>
`;

exports[`styled-components output for a style block with a sibling selector 1`] = `
.namespace .c1 {
  padding: 4px;
}

.namespace .c1 + .c0 {
  margin-left: 8px;
}

<button
  className="c0 c1"
/>
`;

exports[`styled-components output for a style block with interpolated mixins 1`] = `
.namespace .c0 {
  display: block;
  position: relative;
}

.namespace .c0:hover {
  opacity: 1;
}

<div
  className="c0"
/>
`;

exports[`styled-components output for a style block with interpolated selectors 1`] = `
.namespace .c0 {
  position: relative;
}

.namespace .c0 .c1 {
  -webkit-transform: scale(90%);
  -ms-transform: scale(90%);
  transform: scale(90%);
}

.namespace .c2 + .c0,
.namespace .c0 + .sc-dnqmqq {
  margin-right: 12px;
}

<div
  className="c0"
/>
`;

exports[`styled-components output for a style block with no selectors 1`] = `
.namespace .c0 {
  background-color: #333;
}

<div
  className="c0"
/>
`;
